src/main/java/com/google/api/codegen/discovery/Schema.java

```diff
@@ -77,22 +89,35 @@ public static Schema from(DiscoveryNode root, String path) {
     boolean required = root.getBoolean("required");
     Type type = Type.getEnum(root.getString("type"));
 
-    return new AutoValue_Schema(
-        additionalProperties,
-        defaultValue,
-        description,
-        format,
-        id,
-        isEnum,
-        items,
-        location,
-        path,
-        pattern,
-        properties,
-        reference,
-        repeated,
-        required,
-        type);
+    Schema thisSchema =
+        new AutoValue_Schema(
+            additionalProperties,
+            defaultValue,
+            description,
+            format,
+            id,
+            isEnum,
+            items,
+            location,
+            path,
```

Comment:
The addition of `parent` mostly eliminates the need for `path`. I would go ahead and remove it. If you're willing, removing all the boilerplate for path construction can happen in this PR.